Makefile

```diff
@@ -94,3 +95,22 @@ validate-modules:
 tools: ## Install required tools and utilities
 	go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.55.2
 	go install github.com/axw/gocov/gocov@latest
+	go install golang.org/x/tools/cmd/goimports@latest
+
+# Apply go fmt to the codebase
+fmt:
+	go list -f '{{.Dir}}' $(MODULE)/... \
+		| xargs gofmt -s -l -w
+
+goimports:
+	go list -f {{.Dir}} $(MODULE)/... \
+		| xargs goimports -local $(MODULE) -w
```

Comment:
Just curious if both are needed, goimports is an extension of gofmt, it already does everything that gofmt performs.

Reply:
I removed fmt as goimports extends fmt functionality.